### PR TITLE
add support for mac and arm systems to the github action

### PR DESCRIPTION
**Pull request by dyc3**
I've tested this on an M1 mac to make sure it works.

### CODE REVIEW

**Comment by epage** [comment]
Could you elaborate on motivation for this change?

**Comment by dyc3** [comment]
Sure. I was setting up some CI for work, and we blew through the included CI minutes we had. I figured I'd move all of our CI to run on a self hosted runner, running on a mac mini. This was the only action that blew up when trying to run everything on that mac self hosted runner.

I believe this change would also make it possible for raspberry pis to run this action now, but I haven't tested it.

**Comment by epage** [comment]
... your only self-hosted runner is a Mac?  That is the opposite of what I would expect since they have historically been more expensive runners that are harder to provision.  Like github, I've tended to see those kept to a minimum.  I'm assuming this is a shop focused specifically on Apple development??

**Comment by epage** [comment]
We should update https://github.com/crate-ci/typos/blob/master/.github/workflows/test-action.yml to cover at least one of the newly supported arm platforms so we don't regress this

**Comment by dyc3** [comment]
Yeah it's weird. Small startup and we need to do ios builds. I'll update that action.

**Comment by dyc3** [comment]
I've updated the action. The `macos-latest` runner uses M1. See: https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners/about-github-hosted-runners#standard-github-hosted-runners-for-public-repositories

File: .github/workflows/test-action.yml [hunk 1]
```diff
@@ -4,7 +4,10 @@ on: [pull_request]
 jobs:
   shallow:
     name: Spell Check with Typos
-    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest]
```
Comment (epage) [inline .github/workflows/test-action.yml:9]: `macos-latest` is still x64, right?  To ensure we also cover arm, I think we'd need `macos-14`

**Comment by epage** [comment]
Oh, hadn't kept track that `macos-latest` now points to `macos-14`!

**Pull request merged**